### PR TITLE
Feature/omicron tests

### PR DESCRIPTION
**Pull request by aikewoody**
# Description

[comment]: <> (Description of the feature or fix)
[comment]: <> (Include link to Figma designs, if relevant)
[comment]: <> (Include link to Notion ticket, if relevant)

## On what platform have you tested the change?

- [ ] Android - Emulator
- [ ] Android - Physical device
- [x] iOS - Emulator
- [ ] iOS - Physical device

## Out of scope and potential follow-up

[comment]: <> (Are there any related changes that you plan to do in a follow-up PR or voluntarily excluded from the scope?)

## Screenshots

[comment]: <> (Please attach screenshots showing the change, if relevant)

## Checklist (for submission)

- [ ] Analytics/tracking events have been added (if relevant)
- [ ] Added new screen(s) to `src/features/screens.ts` and new Redux slice(s) to `mocks/mockedInitialState.ts` (if relevant)

## Checklist (for reviewers)

- [ ] Checked against Figma designs (if relevant)
- [ ] Checked that data has been successfully saved to the backend (if relevant)

## Checklist (for merger)

- [ ] Make sure the right merge strategy is chosen, commit merge into master and squash merge into develop.

### CODE REVIEW

File: assets/lang/en.json [hunk 1]
```diff
@@ -300,6 +300,18 @@
...
+    "omicron": {
+      "picker-no": "No, I was not contacted by NHS Test & Trace about Omicron",
+      "picker-yes-confirmed": "Yes, I was told it was confirmed Omicron",
+      "picker-yes-suspected": "Yes, I was told it was suspected Omicron",
+      "tag": "New",
+      "title": "Have you been informed by NHS Test & Trace app that it was the Omicron variant?"
+    },
+    "tracking-omicron-modal": {
+      "description": "Help track the new *Omicron variant* by telling us if you were contacted by *NHS Test & Trace* about Omicron with your *recent postive PCR test logged*.\n\nIf you know anyone who has/had *suspected or confirmed Omicron infection*, please also ask them to log it in this app",
+      "tag": "Tracking omicron",
```
Comment (bionara) [inline assets/lang/en.json:313]: should be uppercase O :)

Comment (aikewoody) [reply assets/lang/en.json:313]: The component where this text is used makes the whole string capitalised

File: src/features/covid-tests/questions/CovidTestMechanismQuestion.tsx [hunk 2]
```diff
@@ -282,6 +280,7 @@ CovidTestMechanismQuestion.createDTO = (formData: ICovidTestMechanismData): Part
       is_rapid_test: true,
       mechanism: ECovidTestMechanismOptions.NOSE_OR_THROAT_SWAB,
     }),
+    // @todo: why is the mechanism overwritten here?!
```
Comment (juliatan) [inline src/features/covid-tests/questions/CovidTestMechanismQuestion.tsx:283]: @aikewoody The backend has no concept of "PCR" as a mechanism. The PCR wording in the question is fairly new - historically, mechanism was saved as "nose_throat_swab" etc. When the PCR question was introduced, data science wanted to ensure it was still compatible with historical data, so it was agreed that PCR would be defined as "nose_throat_swab" AND "is_rapid_test" == false.

File: src/features/covid-tests/modals/CovidTestAntibodyInfoModal.tsx [hunk 3]
```diff
@@ -29,7 +29,7 @@ const AntibodyInformation = ({ antibody }: IAntibodyInformationProps) => {
 export const CovidTestAntibodyInfoModal: React.FC<IProps> = (props) => {
   const footerChildren = React.useMemo(
     () => <BrandedButton onPress={props.onRequestClose}>{i18n.t('ok')}</BrandedButton>,
-    [props.onRequestClose, i18n.t('ok')],
```
Comment (juliatan) [inline src/features/covid-tests/modals/CovidTestAntibodyInfoModal.tsx:32]: @aikewoody Please can you double check that this renders correctly for non-UK locales? I think this does need to be included...

File: src/features/covid-tests/screens/CovidTestDetailScreen.tsx [hunk 4]
```diff
@@ -151,106 +148,130 @@ export default function CovidTestDetailScreen(props: TCovidProps) {
...
+            />
+          ) : null}
+
+          <ErrorText>{errorMessage}</ErrorText>
+
+          {!!Object.keys(formikProps.errors).length && formikProps.submitCount > 0 ? (
+            <ValidationError error={i18n.t('validation-error-text')} />
+          ) : null}
+
+          {testId ? <DeleteButton onPress={promptDeleteTest} /> : null}
```
Comment (juliatan) [inline src/features/covid-tests/screens/CovidTestDetailScreen.tsx:241]: @aikewoody It seems like the margins around the submit button and delete link has changed from before.

**Pull request merged**